Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1676,28 +1676,38 @@ class BrowserViewController: UIViewController {
             }
             
             if #available(iOS 14.0, *), let webView = tab?.webView, tab?.temporaryDocument == nil {
-                let createPDFActivity = CreatePDFActivity(webView: webView) { [weak self] pdfData in
-                    guard let self = self else { return }
-                    // Create a valid filename
-                    let validFilenameSet = CharacterSet(charactersIn: ":/")
-                        .union(.newlines)
-                        .union(.controlCharacters)
-                        .union(.illegalCharacters)
-                    let filename = webView.title?.components(separatedBy: validFilenameSet).joined()
-                    let url = URL(fileURLWithPath: NSTemporaryDirectory())
-                        .appendingPathComponent("\(filename ?? "Untitled").pdf")
-                    do {
-                        try pdfData.write(to: url)
-                        let pdfActivityController = UIActivityViewController(activityItems: [url], applicationActivities: nil)
-                        if let popoverPresentationController = pdfActivityController.popoverPresentationController {
-                            popoverPresentationController.sourceView = sourceView
-                            popoverPresentationController.sourceRect = sourceRect
-                            popoverPresentationController.permittedArrowDirections = arrowDirection
-                            popoverPresentationController.delegate = self
+                let createPDFActivity = CreatePDFActivity() {
+                    
+                    webView.createPDF { [unowned self] result in
```

Comment:
Shouldn't use `unowned` in this case even if its unlikely for BVC to dealloc during this time

Reply:
i changed it precisely for this reason, bvc never deallocates, will go back to self, thanks

Reply:
I know its probably not going to in the current architecture, but `createPDF` is an escaping and long-running call so in this case we'd want to use `weak` anyways.